### PR TITLE
relates to #178: insert commands per spec, added oerdered/unordered, better tests

### PR DESCRIPTION
**Pull request by ivansenic**
**What this PR does**:

Adaptation to spec for the `insertOne` & `insertMany`, including test improvements.

* [x] removed option class from `insertOne`  command as per spec no options exists
* [x] added `ordered` option for the `insertMany`
* [x] implemented ordered/unordered insertion
* [x] implemented fail fast for order version
* [x] implemented fail silently for unordered
* improved tests:
   * [x] extended operation tests to ensure all failure modes are tested
   * [x] created/updated tests for command resolvers 
   * [x] created tests for commands validation 
   * [x] several fixes for the integration tests:
      * add unordered tests
      * simulate possible failure modes (using duplicated entries in insert many)
      * assert no errors & data
      * improved readability
* [x] fixed swagger examples

**Which issue(s) this PR fixes**:
Relates to #178.

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperationPage.java [hunk 1]
```diff
@@ -2,21 +2,73 @@
...
+  }
+
+  /** {@inheritDoc} */
   @Override
   public CommandResult get() {
+    // if we have errors, transform
+    if (null != failedIds && !failedIds().isEmpty()) {
+
+      List<CommandResult.Error> errors = new ArrayList<>(failedIds.size());
+      failedIds.forEach((documentId, throwable) -> errors.add(getError(documentId, throwable)));
```
Comment (ivansenic) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperationPage.java:40]: currently for each failed document, I return an error, should we aggregate? if yes how?

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java [hunk 2]
```diff
@@ -13,61 +14,148 @@
...
+                      .map(WritableShreddedDocument::id)
+                      .toList();
+
+              return new InsertOperationPage(
+                  insertedId,
+                  Collections.singletonMap(failed.id(), failFastInsertException.getCause()));
+            })
+
+        // use object identity to resolve to Supplier<CommandResult>
+        .map(i -> i);
```
Comment (ivansenic) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java:94]: can be removed

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java [hunk 3]
```diff
@@ -13,61 +14,148 @@
...
+            doc ->
+                insertDocument(queryExecutor, query, doc)
+
+                    // handle errors fail silent mode
+                    .onItemOrFailure()
+                    .transform((id, t) -> Tuple2.of(doc, t)))
+
+        // then reduce here
+        .collect()
+        .in(InsertOperationPage::new, (agg, in) -> agg.aggregate(in.getItem1().id(), in.getItem2()))
```
Comment (ivansenic) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java:117]: this is not thread safe

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java [hunk 4]
```diff
@@ -13,61 +14,148 @@
...
+        // ensure to construct correctly the information about inserted documents
+        .onFailure(FailFastInsertException.class)
+        .recoverWithItem(
+            e -> {
+              // safe to cast, asserted class in onFailure
+              FailFastInsertException failFastInsertException = (FailFastInsertException) e;
+              WritableShreddedDocument failed = failFastInsertException.document;
+
+              // collect inserted, since it's sequential iterate until failed index
+              int failedIndex = documents().lastIndexOf(failed);
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java:81]: Do we need this logic, since we use concatenate is already ordered and status can be captured with invoke() similar to unordered.

InsertOperationPage response = new InsertOperationPage();
`insertDocument(queryExecutor, query, doc)
                    .onItemOrFailure()
                    .invoke(
                        (documentId, error) -> {
                          response.aggregate(doc.id(), error);
                        }))`

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java [hunk 5]
```diff
@@ -13,61 +14,148 @@
...
+                    // handle errors fail silent mode
+                    .onItemOrFailure()
+                    .transform((id, t) -> Tuple2.of(doc, t)))
+
+        // then reduce here
+        .collect()
+        .in(InsertOperationPage::new, (agg, in) -> agg.aggregate(in.getItem1().id(), in.getItem2()))
+
+        // use object identity to resolve to Supplier<CommandResult>
+        .map(i -> i);
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java:120]: Why is this needed in() would have already return Uni< InsertOperationPage>

**Review by maheshrajamani — APPROVED** [review]

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java (hunk 4 above)
Comment (ivansenic) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java:81]: Hmm but how can I reduce? Where is the `response` coming from?

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java (hunk 5 above)
Comment (ivansenic) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java:120]: Does not compile. The method signature must be 

`Uni<? extends Supplier<CommandResult>>` 

so that we can return the `Uni<InsertOperationPage>`.

I wanted to refactor, but it's not a small change, should be changed in all sub-classes and tests.. Do you think it's worth it?

**Comment by ivansenic** [comment]
@maheshrajamani can you answer to the questions I posted?

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java (hunk 4 above)
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java:81]: @ivansenic 
`InsertOperationPage response = new InsertOperationPage();
    return Multi.createFrom()
        .items(documents.stream())
        .onItem()
        .transformToUniAndConcatenate(
            doc ->
                insertDocument(queryExecutor, query, doc)
                    .onItemOrFailure()
                    .invoke(
                        (documentId, error) -> {
                          response.aggregate(doc.id(), error);
                        }))
        .collect()
        .last()
        .onItemOrFailure()
        .transform(
            (id, error) -> {
              return response;
            });`

Comment (ivansenic) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java:81]: Hmm, I don't like those side effects, especially when I am using a non-thread-safe structure.. Let me see if I can optimize this..

Comment (ivansenic) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java:81]: @maheshrajamani this is the bug from the demo, I can not use last index on a document, as if first fails, I would assume that the first was written.. Need to change this :)
